app/src/controllers/bucket.js

```diff
@@ -63,6 +65,34 @@ const controller = {
     return exposedHeaders;
   },
 
+  /**
+   * @function _validateCredentials
+   * Gaurd against creating or update a bucket with invalid creds
+   * @param {object} requestBody The body of the request
+   */
+  async _validateCredentials(requestBody, res) {
+    try {
+      const bucketSettings = {
+        accessKeyId: requestBody.accessKeyId,
+        bucket: requestBody.bucket,
+        endpoint: requestBody.endpoint,
+        key: requestBody.key,
+        region: DEFAULTREGION,
```

Comment:
Region will need to be accepted as an optional requestBody parameter at some point; defaulting to DEFAULTREGION if undefined.
```js
region: requestBody.region || DEFAULTREGION,
```

Reply:
Done